platforms/common/spi.cpp

```diff
@@ -123,10 +123,14 @@ bool SPIBusIterator::next()
 			case FilterType::InternalBus:
 				if (!bus_data.is_external) {
 					if (_bus == bus_data.bus || _bus == -1) {
-						// find device id
+						// if chipselect < 0, it's not defined
+						bool use_cs = _chipselect >= 0;
+
+						// find device
 						for (int i = _bus_device_index + 1; i < SPI_BUS_MAX_DEVICES; ++i) {
 							if (PX4_SPI_DEVICE_ID == PX4_SPIDEVID_TYPE(bus_data.devices[i].devid) &&
-							    _devid_driver_index == bus_data.devices[i].devtype_driver) {
+							    _devid_driver_index == bus_data.devices[i].devtype_driver &&
+							    (!use_cs || _chipselect == (int16_t)(bus_data.devices[i].cs_gpio & GPIO_PIN_MASK))) {
```

Comment:
use_cs could be directly replaced by (_chipselect >=0) here. sort of useless variable...

Reply:
fixed